source/common/ur_util.hpp

```diff
@@ -293,4 +293,6 @@ inline ur_result_t exceptionToResult(std::exception_ptr eptr) {
     }
 }
 
+template <class> inline constexpr bool ur_always_false_t = false;
```

Comment:
Defer the evaluation of this until it is actually instantiated. using `static_assert(false, "blah");` will always fail.